src/libsync/propagatedownload.cpp

```diff
@@ -502,9 +534,33 @@ void PropagateDownloadFile::slotGetFinished()
 {
     propagator()->_activeJobList.removeOne(this);
 
-    GETFileJob *job = qobject_cast<GETFileJob *>(sender());
+    GETJob *job = qobject_cast<GETJob *>(sender());
     ASSERT(job);
 
+    SyncFileItem::Status status = job->errorStatus();
+
+    // Needed because GETFileZsyncJob may emit finishedSignal without any further network activity
+    if (!job->reply()) {
+        if (status == SyncFileItem::Success) {
+            _tmpFile.close();
+            _tmpFile.flush();
+            downloadFinished();
+            return;
+        }
+
+        FileSystem::remove(_tmpFile.fileName());
```

Comment:
Again, this might break resuming.

Reply:
Again, same answer as before, if we error'd when parsing zsync then this file was simply created and contains 0 bytes.